File: src/test/java/fr/arsenelapostolet/professor/server/unit/application/FakeStudentRepository.java
package fr.arsenelapostolet.professor.server.unit.application;

import fr.arsenelapostolet.professor.server.core.entities.Student;
import fr.arsenelapostolet.professor.server.core.repositories.StudentRepository;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

public class FakeStudentRepository implements StudentRepository {

    private final Map<String, Student> data = new HashMap<>();

    @Override
    public Set<Student> saveStudents(Set<Student> students) {
        data.putAll(students.stream().collect(Collectors.toMap(Student::getId, student -> student)));
        return students;
    }

    public Map<String, Student> getData() {
        return data;
    }
}

File: app/src/test/kotlin/fr/arsenelapostolet/professor/fakes/FakeStudentRepository.kt
package fr.arsenelapostolet.professor.fakes

import fr.arsenelapostolet.professor.core.entities.Student
import fr.arsenelapostolet.professor.core.repositories.StudentRepository

class FakeStudentRepository(data: Map<String, Student> = emptyMap()) : StudentRepository,
    FakeRepository<Student>(data) {

    override fun saveStudents(students: Set<Student>): Set<Student> {
        for (student in students) {
            data[student.id] = student
        }

        return students.toSet();
    }

    override fun getAllStudents(): Set<Student> = data.values.toSet();
}